releasenotes/notes/improve-hls-pass-620389dde24e9707.yaml

```diff
@@ -0,0 +1,6 @@
+---
+features_transpiler:
```

Comment:
From the tests it seemed that the output changed as well, we should add an `upgrade` releasenote here too. Though there is no way to still achieve the old behavior, or is there?

Reply:
I don't understand this comment: in the past we synthesized AnnotatedOperations as a part of the HighLevelSynthesis transpiler pass itself, and now we are doing it as a part of the default plugin for AnnotatedOperations- but these details should not impact users in any way, right?

Of course, with general improvements to HighLevelSynthesis + plugins, the circuits produced by `transpile` may change, but I think that's expected.

Reply:
> Of course, with general improvements to HighLevelSynthesis + plugins, the circuits produced by transpile may change, but I think that's expected.

Such a change in the default behavior should be documented though, I think, since users could rely on the previous behavior. If it's a more efficient decomposition using the same set of gates it is maybe less important, but by looking at the tests it seems that here the wrapping of the circuit has changed, or no? 🤔 

<img width="697" alt="image" src="https://github.com/user-attachments/assets/976cde67-b1d0-4469-9633-41d89fc9b359" />

In this case I think we should leave a note, as users could've relied on the previous wrapping in some way they used "decompose" or counted the operations

Reply:
Ah, right, done in [e6dffe8](https://github.com/Qiskit/qiskit/pull/13813/commits/e6dffe8e5736d70c15df153f7f68edce42ea2c45).